src/main/kotlin/exercise4/task2/TransactionalBankAccount.kt

```diff
@@ -78,8 +78,136 @@ import java.util.*
  *      ```
  */
 
+class TransactionalBankAccount(accountNumber: String,
+                               accountHolderName: String,
+                               balance: Double = 0.0
+) : BankAccount(accountNumber, accountHolderName, balance) {
 
+    private val transactions = mutableListOf<Transaction>()
 
+    override fun deposit(amount: Double): Boolean {
+        val success = super.deposit(amount)
+        recordTransaction(amount, success, TransactionType.DEPOSIT)
+        return success
+    }
+
+    override fun withdraw(amount: Double): Boolean {
+        val success = super.withdraw(amount)
+        recordTransaction(amount, success, TransactionType.WITHDRAWAL)
+        return success
+    }
+
+    private fun recordTransaction(amount: Double, success: Boolean, transactionType: TransactionType) {
+        val transaction = Transaction(
+            LocalDateTime.now(),
+            transactionType,
+            amount,
+            if (transactionType == TransactionType.DEPOSIT) {
+                if (amount > 0)
+                    getBalance() + amount
+                else {
+                    getBalance()
+                }
+            } else {
+                getBalance() - amount
+            },
+            getBalance(),
+            if (success)
+                TransactionStatus.SUCCESS
+            else
+                TransactionStatus.FAILURE
+        )
+        transactions.add(transaction)
+    }
+
+    override fun getBalance(): Double = super.getBalance()
```

Comment:
There is no need to override the method, since it just delegates the call to the supper class.